src/Nethermind/Nethermind.Synchronization/FastSync/TreeSync.cs

```diff
@@ -850,6 +854,11 @@ private void LogRequestInfo(List<StateSyncItem> requestHashes)
             }
         }
 
+        private void OnStateRangesFinished(object? sender, SnapSyncEventArgs e)
+        {
+            _data.SetSnapSyncData(e.StateRangesFinished, e.StateRangesSyncedBytes);
```

Comment:
Recommend adding `DetailedProgress` into `ProgressTracker` instead of `ProgressTracker` into `TreeSync`, and setting this there instead.

Reply:
Wouldn't it get even more coupled? `DetailedProgress` is private for `TreeSync` and `ProgressTracker` is created during `InitializeNetwork`

Reply:
Ow... did not expect. I would have expected DetailedProgress to be not state sync specific...